Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.cpp

```diff
@@ -260,24 +289,55 @@ namespace AzToolsFramework::Prefab
         // Refresh the read-only cache, if the interface is initialized.
         if (m_readOnlyEntityQueryInterface)
         {
-            EntityIdList containerEntities;
+            EntityIdList entities;
 
             if (previousFocusedInstance.has_value())
             {
-                containerEntities.push_back(previousFocusedInstance->get().GetContainerEntityId());
+                previousFocusedInstance->get().GetEntities(
+                    [&](AZStd::unique_ptr<AZ::Entity>& entity) -> bool
+                {
+                    entities.push_back(entity->GetId());
+                    return true;
+                }
+                );
+                entities.push_back(previousFocusedInstance->get().GetContainerEntityId());
             }
-            containerEntities.push_back(focusedInstance->get().GetContainerEntityId());
+            entities.push_back(focusedInstance->get().GetContainerEntityId());
+            focusedInstance->get().GetEntities(
+                [&](AZStd::unique_ptr<AZ::Entity>& entity) -> bool
+            {
+                entities.push_back(entity->GetId());
+                return true;
+            });
 
-            m_readOnlyEntityQueryInterface->RefreshReadOnlyState(containerEntities);
+            m_readOnlyEntityQueryInterface->RefreshReadOnlyState(entities);
         }
 
         // Refresh path variables.
         RefreshInstanceFocusPath();
 
         // Open all container entities in the new path.
         SetInstanceContainersOpenState(m_rootAliasFocusPath, true);
+        // Set open state on all nested instances in the new focus subtree based on edit scope.
+        SetInstanceContainersOpenStateOfAllDescendantContainers(
+            GetInstanceReference(m_rootAliasFocusPath), m_prefabEditScope == PrefabEditScope::SHOW_NESTED_INSTANCES_CONTENT);
+
+        PrefabFocusNotificationBus::Broadcast(&PrefabFocusNotifications::OnPrefabFocusChanged,
+            previousFocusedInstance->get().GetContainerEntityId(), focusedInstance->get().GetContainerEntityId());
 
-        PrefabFocusNotificationBus::Broadcast(&PrefabFocusNotifications::OnPrefabFocusChanged);
+        // Force propagation both the previous and the new focused instances to ensure they are represented correctly.
+        if (m_instanceUpdateExecutorInterface)
+        {
+            auto previouslyFocusedInstance = GetInstanceReference(previousContainerRootAliasPath);
+
+            // The most common operation is focusing a prefab instance nested in the currently focused instance.
+            // Queuing the previous focus before the new one saves some time in the propagation loop on average.
+            if (previouslyFocusedInstance.has_value())
+            {
+                m_instanceUpdateExecutorInterface->AddInstanceToQueue(previouslyFocusedInstance);
+            }
+            m_instanceUpdateExecutorInterface->AddInstanceToQueue(focusedInstance);
```

Comment:
Note: This change https://github.com/o3de/o3de/pull/11120 from @forkercat is already reintroducing this same mechanism to propagate the focused instance. If this PR goes in after theirs, make sure to pull it and resolve the merge first!